pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,194 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+
+	"kubevirt.io/client-go/kubecli"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+)
+
+type IPAMClaimParams struct {
+	claimName   string
+	networkName string
+}
```

Comment:
Thinking again about this we don't need the struct if we do the `fmt.Sprintf("%s.%s", vmiName, network.Name)` inside `ComposeIPAMClaim` since this is the right place.

Reply:
ComposeIPAMClaim is not the only client
it means that all GenerateMultusCNIAnnotation, updateMultusAnnotation and GenerateMultusCNIAnnotationFromNameScheme
will need to get vmi.Name in addition to the new map[string]string that you suggest which will have only `networkName`, and imo it less nicer to have both the map and then need to build inside the string, and it adds another parameter

Note, lets please first decide a direction, and then refer to the other comments, because almost all of them around it (beside one that i added comment to already)